src/Workspaces/CSharp/Portable/CodeGeneration/OperatorGenerator.cs

```diff
@@ -96,6 +96,28 @@ internal static class OperatorGenerator
                 parameterList: ParameterGenerator.GenerateParameterList(method.Parameters, isExplicit: false, options: options),
                 body: hasNoBody ? null : StatementGenerator.GenerateBlock(method),
                 semicolonToken: hasNoBody ? SyntaxFactory.Token(SyntaxKind.SemicolonToken) : new SyntaxToken());
+
+            operatorDecl = UseExpressionBodyIfDesired(options, operatorDecl, parseOptions);
+            return operatorDecl;
+        }
+
+        private static OperatorDeclarationSyntax UseExpressionBodyIfDesired(
+            CodeGenerationOptions options, OperatorDeclarationSyntax operatorDeclaration, ParseOptions parseOptions)
+        {
+            if (operatorDeclaration.ExpressionBody == null)
+            {
+                var expressionBodyPreference = options.Options.GetOption(CSharpCodeStyleOptions.PreferExpressionBodiedMethods).Value;
+                if (operatorDeclaration.Body.TryConvertToArrowExpressionBody(
+                        operatorDeclaration.Kind(), parseOptions, expressionBodyPreference,
+                        out var expressionBody, out var semicolonToken))
+                {
+                    return operatorDeclaration.WithBody(null)
```

Comment:
I'm suprised `WithExpressionBody(nonnull)` doesn't already do the `WithBody(null)` portion of this.

Reply:
> WithExpressionBody

Syntax apis don't make any changes you didn't ask for.  It has no semantic understnading of hte relation here.  it will gladly let you make `F() => bar { }`